tests/std/tests/Dev11_0836436_get_time/test.cpp

```diff
@@ -760,3 +766,56 @@ void test_locale_chinese() {
     assert(read_date_locale(L"2020-\x5341\x4e8c\x6708-31", "zh-CN") == make_tuple(31, 11, 120));
     assert(read_date_locale(L"2020-\x0031\x0032\x6708-31", "zh-CN") == make_tuple(31, 11, 120));
 }
+
+void test_invalid_parameter_handler(const wchar_t* const expression, const wchar_t* const function,
+    const wchar_t* const file, const unsigned int line, const uintptr_t reserved) {
+    (void) expression;
+    (void) reserved;
+
+    static int num_called = 0;
+    if (num_called++ > 10) {
+        wprintf(
+            L"Test Failed: Invalid parameter handler was called over 10 times by %s in %s:%u\n", function, file, line);
+        exit(1);
+    }
+}
+
+void test_invalid_argument() {
+    _set_invalid_parameter_handler(test_invalid_parameter_handler);
+
+    time_t t = time(nullptr);
+    tm currentTime;
+    localtime_s(&currentTime, &t);
+
+    {
+        wstringstream wss;
+        wss << put_time(&currentTime, L"%Y-%m-%d-%H-%M-%s");
+        assert(wss.rdstate() == ios_base::badbit);
+    }
+
+    {
+        stringstream ss;
+        ss << put_time(&currentTime, "%Y-%m-%d-%H-%M-%s");
+        assert(ss.rdstate() == ios_base::badbit);
+    }
+}
+
+void test_buffer_resizing() {
+    time_t t = time(nullptr);
+    tm currentTime;
+    localtime_s(&currentTime, &t);
+
+    {
+        wstringstream wss;
+        wss.imbue(locale("ja_JP"));

```

Comment:
We've had problems with underscored locale names in our internal test infrastructure (my vague understanding is that a few of the internal test machines use Server 2016 which didn't understand underscores). Everywhere else, including above in this test (e.g. line 767), we use hyphenated locales. I believe we should do so here for consistency, and to avoid problems when porting this PR internally.

Occurs twice in this file.